web3js-ext/test/role_based_key_feepayer.spec.ts

```diff
@@ -0,0 +1,149 @@
+    });
+
+    // Test Case 2: Attempting to sign a regular transaction with RoleFeePayer key (should fail)
+    it("2. Attempting to sign a regular transaction with RoleFeePayer key (failure test)", async function () {
+        const valueTx = {
+            type: TxType.ValueTransfer,
+            from: generateTemporaryAccount().address,
+            to: generateTemporaryAccount().address,
+            value: toPeb("0.01"),
+            gasLimit: 100000
+        };
+
+        try {
+            const signedTx = await roleFeePayerAccount.signTransaction(valueTx);
+            assert.fail("RoleFeePayer key should not sign a ValueTransfer transaction.");
+        } catch (error: any) {
+            assert.isTrue(true, "Error occurred as expected due to role mismatch");
+        }
+    });
+
+    // Test Case 3: Fee Delegated transaction signed by RoleFeePayer key (should succeed)
+    it("3. Fee Delegated transaction signed by RoleTransaction and RoleFeePayer keys", async function () {
+        // Feedback 4. Create a new user account (different user than FeePayer)
+        const userAccount = generateTemporaryAccount();
+
+        const feeDelegatedTx = {
+            type: TxType.FeeDelegatedValueTransfer,
+            from: userAccount.address,
+            to: generateTemporaryAccount().address,
+            value: toPeb("0.01"),
+            gasLimit: "100000", 
+            gasPrice: "25000000000", 
+            nonce: "0x0",
+            chainId: "0x1001" 
+        };
+
+        // Feedback 4. User signs transaction first
+        const signedTxByUser = await userAccount.signTransaction(feeDelegatedTx);
+        assert.isNotNull(signedTxByUser.rawTransaction, "Transaction should be signed by User");
+
+        // Feedback 4. FeePayer adds an additional signature
+        try {
+            const signedTxByFeePayer = await roleFeePayerAccount.signTransaction({
+                senderRawTransaction: signedTxByUser.rawTransaction,
+                gasPrice: "25000000000",
+                gasLimit: "100000" 
+            });
+            assert.isNotNull(signedTxByFeePayer.rawTransaction, "FeePayer should sign the transaction");
+        } catch (error: any) {
+            console.log("error is : ", error.message);
+            assert.fail("FeePayer failed to sign the already signed transaction.");
+        }
+    });
+
+    // Test Case 4: Attempting to sign Fee Delegated transaction with RoleTransaction key (should fail)
+    it("4. Attempting to sign Fee Delegated transaction with RoleTransaction key (should fail)", async function () {
+        const userAccount = generateTemporaryAccount();
+
+        const feeDelegatedTx = {
+            type: TxType.FeeDelegatedValueTransfer,
+            from: userAccount.address,
+            to: generateTemporaryAccount().address,
+            value: toPeb("0.01"),
+            gasLimit: 100000,
+            gasPrice: "25000000000", 
+            nonce: "0x0",
+            chainId: "0x1001"
```

Comment:
Thanks @kjeom , thanks to your advice, I solved the "insufficient funds" error.

**[Feedback]**
1. Manually input transaction fields
- Changed to manually input gasPrice, gasLimit, nonce, chainId to solve the "insufficient funds for transfer" error.
- This prevents RPC calls such as eth_estimateGas, eth_getTransactionCount, etc., so that the problem does not occur even on test accounts without Kaia balance.

And this solved the "insufficient funds for transfer" error.

**[Unexpected Observations]**
But unfortunately? The unit test succeeded, so I was worried a lot. (Because if there is a bug, it should not succeed)

Test case 3: Expected bug does not appear
- I expected it to fail due to role mismatch when signing with Role-based FeePayer key in FeeDelegatedValueTransfer transaction, but it succeeded.

Test Case 4: Unexpected Success
- I expected it to fail when signing a FeeDelegatedValueTransfer transaction with roleTransactionAccount, but it succeeded.

**[Result & Question]**
I was wondering why it succeeded, and I thought that maybe it was a bug.

- This is because the signTransaction method in Kaia's web3js-ext does not perform role-based verification.
- Or role-based verification is only performed at the network level in the blockchain node.
- So it seems that the signTransaction method simply generates a signature and does not check the role restriction, so it is signed normally as above.

Reply:
@minminkikiki If you try RLP decoding below code, you can check the feePayer address is different address that is originally intended.
```js
const { KlaytnTxFactory } = require("@kaiachain/web3js-ext");
const decoded = KlaytnTxFactory.fromRLP(signedTx.rawTransaction);
console.log(decoded)
```